hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/handlers/TssUtils.java

```diff
@@ -123,13 +125,27 @@ public static List<TssMessage> getTssMessages(List<TssMessageTransactionBody> va
      */
     public static Map<Long, Long> computeNodeShares(
             @NonNull final List<RosterEntry> rosterEntries, final long maxSharesPerNode) {
+        final var weights = new LinkedHashMap<Long, Long>();
+        rosterEntries.forEach(entry -> weights.put(entry.nodeId(), entry.weight()));
+        return computeSharesFromWeights(weights, maxSharesPerNode);
```

Comment:
Minor:
1. The new map consumes extra memory and doesn't seem to provide any extra benefit as opposed to using the `RosterEntry::weight` directly as the old code did.
2. Modifying external collections from lambdas is generally an anti-pattern because the code would break the moment someone makes the stream parallel.

Both the above issues would be amplified if/when the size of the roster grows significantly as more nodes join the network.

Reply:
Thanks Anthony, re (1) I wanted to reuse the node's precise shares-from-weights calculation [here](https://github.com/hashgraph/hedera-services/blob/16309-tss-txn-sim/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/tss/RekeyScenarioOp.java#L420) when setting the test expectations; re (2) this isn't a stream, just a `forEach()`.

Reply:
Fair.